pkg/apis/core/types.go

```diff
@@ -5788,3 +5796,470 @@ type PortStatus struct {
+	// allocated.
+	Allocation *AllocationResult
+
+	// ReservedFor indicates which entities are currently allowed to use
+	// the claim. Usually those are Pods, but other objects are also
+	// possible as long as they exist as resources in the apiserver. The
+	// ResourceClaim controller in the kube-controller-manager checks for
+	// stale entries that reference non-existent objects and removes them.
+	//
+	// A scheduler must add a Pod that it is scheduling. This must be done
+	// in an atomic ResourceClaim update because there might be multiple
+	// schedulers working on different Pods that compete for access to the
+	// same ResourceClaim, the ResourceClaim might have been marked for
+	// deletion, or even been deallocated already.
+	//
+	// kubelet will check this before allowing a Pod to run because a a
+	// user might have selected a node manually without reserving resources
+	// or a scheduler might have missed that step, for example because it
+	// doesn't support dynamic resource allocation or the feature was
+	// disabled.
+	//
+	// The maximum size is 32 (= [ResourceClaimReservedForMaxSize]). This
+	// is an artificial limit to prevent a completely unbounded field in
+	// the API.
+	ReservedFor []ResourceClaimUserReference
+
+	// DeallocationRequested gets set by the scheduler when it detects the
+	// situation where pod scheduling cannot proceed because some claim was
+	// allocated for a node that cannot provide some other required
+	// resource.
+	//
+	// The driver then needs to deallocate this claim and the scheduler
+	// will try again.
+	//
+	// While DeallocationRequested is set, no new users may be added to
+	// ReservedFor.
+	DeallocationRequested bool
+}
+
+// ReservedForMaxSize is the maximum number of entries in
+// [ResourceClaimStatus.ReservedFor].
+const ResourceClaimReservedForMaxSize = 32
+
+// AllocationResult contains attributed of an allocated resource.
+type AllocationResult struct {
+	// ResourceHandle contains arbitrary data returned by the driver after a
+	// successful allocation. This is opaque for
+	// Kubernetes. Driver documentation may explain to users how to
+	// interpret this data if needed.
+	//
+	// Resource drivers can use this to store some data directly or
+	// cross-reference some other place where information is stored.
+	// This data is guaranteed to be available when a Pod is about
+	// to run on a node, in contrast to the ResourceClass which
+	// may have been deleted in the meantime.
+	//
+	// The maximum size of this field is 16KiB (= [ResourceHandleMaxSize]).
+	ResourceHandle string
+
+	// This field will get set by the resource driver after it has
+	// allocated the resource driver to inform the scheduler where it can
+	// schedule Pods using the ResourceClaim.
+	//
+	// Node-local resources can use the `kubernetes.io/hostname` label
+	// to select a specific node.
+	//
+	// Setting this field is optional. If nil, the resource is available
```

Comment:
```suggestion
	// Setting this field is optional. If null, the resource is available
```

We use the name in YAML in these comments.

Reply:
Changed.